projects/admin/src/app/migration/deduplication/record/brief-view/migration-data-deduplication/migration-data-deduplication.component.ts

```diff
@@ -0,0 +1,229 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { Component, computed, inject, input, OnInit, output } from '@angular/core';
+import { RecordService } from '@rero/ng-core';
+import { MessageService } from 'primeng/api';
+
+@Component({
+  selector: 'admin-migration-data-deduplication',
+  templateUrl: './migration-data-deduplication.component.html',
+  styleUrl: './migration-data-deduplication.component.scss',
+})
+export class MigrationDataDeduplicationBriefComponent implements OnInit {
+  // services
+  protected toastService: MessageService = inject(MessageService);
+  protected recordService: RecordService = inject(RecordService);
+
+  // current record
+  record = input<any>();
+
+  // need a result list refresh
+  refresh = output<boolean>();
+
+  // current ILS pid
+  ilsPid = null;
+
+  // toggle in place edit ils pid
+  isInplaceActive = false;
+
+  // current candidate
+  currentCandidate = null;
+
+  // current candidate index
+  currentCandidateIndex = null;
+
+  // message logs server
+  messages = computed((): { severity: string; detail: string }[] => this.getMessages());
+
+  // badge color based on status
+  severityTag = computed((): string => this.getSeverityBadge());
+
+  //** OnInit hook */
+  ngOnInit(): void {
+    let ilsPid = this.record()?.metadata?.deduplication?.ils_pid;
+    // get value from the backend if it exists
+    if (ilsPid == null && this.candidates.length > 0) {
+      ilsPid = this.candidates[0].json.pid;
+    }
+    // display the current candidate if exists
+    this.updateCurrentCandidate(ilsPid);
+  }
+
+  /**
+   * Get the backend log messages from the record.
+   *
+   * @returns list of messages on primeng format
+   */
+  getMessages(): { severity: string; detail: string }[] {
+    const messages = [];
+    if (this.record()?.metadata?.deduplication?.logs) {
+      ['info', 'warning', 'error'].map((field) => {
+        const log = this.record()?.metadata?.deduplication?.logs[field];
+        if (log) {
+          messages.push({
+            severity: field == 'warning' ? 'warn' : field,
+            detail: log.join('<br/>'),
+          });
+        }
+      });
+    }
+    return messages;
+  }
+
+  /**
+   * Get the badge color from the status.
+   *
+   * @returns the prime ng severity badge value
+   */
+  getSeverityBadge(): string {
+    let severity = 'secondary';
+    const status = this.record()?.metadata?.deduplication?.status;
+    if (status === 'match') {
+      severity = 'success';
+    }
+    if (status.endsWith('check')) {
+      severity = 'warning';
+    }
+    return severity;
```

Comment:
This code exists in several components. Would it be possible to centralize it?

Reply:
I put the logic into the template. An alternate approach use a custom pipe, but as the logic is deferent for each component (value of the status), the template approach is preferred.